packages/datadog-instrumentations/src/child_process.js

```diff
@@ -0,0 +1,152 @@
+  if (Array.isArray(args[1])) {
+    childProcessInfo.command = childProcessInfo.command + ' ' + args[1].join(' ')
+    if (args[2] != null && typeof args[2] === 'object') {
+      childProcessInfo.options = args[2]
+    }
+  } else if (args[1] != null && typeof args[1] === 'object') {
+    childProcessInfo.options = args[1]
+  }
+  if (shell || childProcessInfo.options?.shell === true || typeof childProcessInfo.options?.shell === 'string') {
+    childProcessInfo.shell = true
+  } else {
+    childProcessInfo.shell = false
+  }
+
+  return childProcessInfo
+}
+
+function wrapChildProcessSyncMethod (shell = false) {
+  return function wrapMethod (childProcessMethod) {
+    return function () {
+      if (!childProcessChannel.start.hasSubscribers || arguments.length === 0) {
+        return childProcessMethod.apply(this, arguments)
+      }
+
+      const childProcessInfo = normalizeArgs(arguments, shell)
+
+      return childProcessChannel.traceSync(
+        childProcessMethod,
+        {
+          command: childProcessInfo.command,
+          shell: childProcessInfo.shell
+        },
+        this,
+        ...arguments)
+    }
+  }
+}
+
+function wrapChildProcessCustomPromisifyMethod (customPromisifyMethod, shell) {
+  return function () {
+    if (!childProcessChannel.start.hasSubscribers || arguments.length === 0) {
+      return customPromisifyMethod.apply(this, arguments)
+    }
+
+    const childProcessInfo = normalizeArgs(arguments, shell)
+
+    return childProcessChannel.tracePromise(
+      customPromisifyMethod,
+      {
+        command: childProcessInfo.command,
+        shell: childProcessInfo.shell
+      },
+      this,
+      ...arguments)
+  }
+}
+
+function wrapChildProcessAsyncMethod (shell = false) {
+  return function wrapMethod (childProcessMethod) {
+    function wrappedChildProcessMethod () {
+      if (!childProcessChannel.start.hasSubscribers || arguments.length === 0) {
+        return childProcessMethod.apply(this, arguments)
+      }
+
+      const childProcessInfo = normalizeArgs(arguments, shell)
+
+      const innerResource = new AsyncResource('bound-anonymous-fn')
+      return innerResource.runInAsyncScope(() => {
+        childProcessChannel.start.publish({ command: childProcessInfo.command, shell: childProcessInfo.shell })
+
+        const childProcess = childProcessMethod.apply(this, arguments)
+        if (childProcess) {
+          let errorExecuted = false
+
+          childProcess.on('error', (e) => {
+            errorExecuted = true
+            childProcessChannel.error.publish(e)
+          })
+
+          childProcess.on('close', (code) => {
+            code = code || 0
+            if (!errorExecuted && code !== 0) {
+              childProcessChannel.error.publish()
+            }
+            childProcessChannel.asyncEnd.publish({
+              command: childProcessInfo.command,
+              shell: childProcessInfo.shell,
+              result: code
+            })
+          })
+        }
```

Comment:
something like that?

```suggestion
      const { command, shell } = normalizeArgs(arguments, shell)

      const innerResource = new AsyncResource('bound-anonymous-fn')
      return innerResource.runInAsyncScope(() => {
        childProcessChannel.start.publish({ command, shell })

        const childProcess = childProcessMethod.apply(this, arguments)
        if (childProcess) {
          let errorExecuted = false

          childProcess.on('error', (e) => {
            errorExecuted = true
            childProcessChannel.error.publish(e)
          })

          childProcess.on('close', (code) => {
            code = code || 0
            if (!errorExecuted && code !== 0) {
              childProcessChannel.error.publish()
            }
            childProcessChannel.asyncEnd.publish({
              command,
              shell,
              result: code
            })
          })
        }
```